e2e/nomostest/testing/ntb.go

```diff
@@ -101,8 +102,52 @@ func (t *FakeNTB) Name() string {
 func (t *FakeNTB) Helper() {
 }
 
-// Cleanup is a no-op function.
-func (t *FakeNTB) Cleanup(func()) {
+// Cleanup registers a cleanup function to call at the end of the test suite.
+func (t *FakeNTB) Cleanup(f func()) {
```

Comment:
Does this mean we used to just skip all Cleanups on kind clusters?
That might explain why it was sometimes faster.

Reply:
Not exactly, this `FakeNTB` is used as an abstraction for shared test environments whose lifecycle is the entire test suite rather than per-test. Before this change we only really used this for GKE and any cleanup functions registered while creating the shared test environment would be ignored. There's a separate [Cleanup](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/e2e/nomostest/clean.go#L69) function that currently gets called for shared test environments on GKE.

The way we currently clean up the Kind clusters is by registering a cleanup function, so supporting cleanup functions with FakeNTB seemed like a straightforward solution to me (the only difference being FakeNTB cleanup gets invoked by us at the end of the test suite).